Assets/HotmaxEditorVR/loadPropsMenu.cs

```diff
@@ -0,0 +1,288 @@
+    private stateManager _stateManagerMutatorRef;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+
+    //--ref to props scene object
+    private GameObject propParent;
+
+    //--menu prop tile parent and child
+    public GameObject propTiles;
+
+    //--Lists to handle getting unique props in menu
+    private List<string> propName = new List<string>();
+    private List<GameObject> propObject = new List<GameObject>();
+    private List<List<GameObject>> chunkedList = new List<List<GameObject>>();
+
+    //--if false, disabled initial props on stage
+    public bool propIsInStageAlready = false;
+
+    private int currentPage;
+
+    //TODO: for some reason the steamVR trigger event listener won't attach in enable or Awake. works in Start...
+    private void OnEnable()
+    {
+        Invoke("waitToLoadTriggerEvent", .001f);
+        stateManager.editorModeEvent += updateEditorMode;
+    }
+
+    void waitToLoadTriggerEvent()
+    {
+        inputManager.trackedController2.TriggerClicked += triggerClicked;
+    }
+
+
+    private void OnDisable()
+    {
+        inputManager.trackedController2.TriggerClicked -= triggerClicked;
+
+        stateManager.editorModeEvent -= updateEditorMode;
+    }
+
+    void updateEditorMode(stateManager.editorModes value)
+    {
+        _editorMode = value;
+
+        if (_editorMode == stateManager.editorModes.spawnMenuMode)
+        {
+            toggleMenuComponents(true);
+
+            //--make menu face the HMD
+            GameObject menu = transform.parent.transform.gameObject;
+            GameObject vrCam = init.vrCamera;
+
+            Vector3 forwardAmount = vrCam.transform.position + (vrCam.transform.forward * 2.15f);
+            Quaternion rotationAmount = vrCam.transform.rotation;
+
+            menu.transform.position = new Vector3(forwardAmount.x, vrCam.transform.position.y - 1, forwardAmount.z);
+
+            menu.transform.rotation = rotationAmount;
+            menu.transform.localEulerAngles = new Vector3(0, menu.transform.localEulerAngles.y + 180, 0);
+        }
+        else
+        {
+            toggleMenuComponents(false);
+        }
+    }
+
+    void Start()
+    {
+        _stateManagerMutatorRef = GameObject.FindObjectOfType(typeof(stateManager)) as stateManager;
+
+        propParent = GameObject.Find("props");
+
+        //get the names of all props in props
+        for (int i = 0; i < propParent.transform.childCount; i++)
+        {
+            propName.Add(propParent.transform.GetChild(i).gameObject.name);
+        }
+
+        //weed out props that have the same name
+        propName = propName.Distinct().ToList();
+
+        //copy the game objects with the unique names to a gameObject list 
+        for (int i = 0; i < propName.Count; i++)
+        {
+            propObject.Add(GameObject.Find(propName[i]));
+        }
+
+        //set all game objects in props inactive if they are not in the stage
+        if (!propIsInStageAlready)
```

Comment:
I don't understand this section - there is a public bool on the menu class that on start drives some prop.active behavior?

Reply:
I did this for ease of use purposes. The objects in "props" have to active initially to be read, this deactivates them so the scene starts empty.

Reply:
hmm ok, i would make a note that says `TODO: remove when props are received from PropClass`

Reply:
got it